scalameta/semantic/src/main/scala/scala/meta/semantic/v1/Symbol.scala

```diff
@@ -19,31 +21,78 @@ import scala.meta.internal.semantic.v1._
 @root trait Symbol extends Optional {
   def syntax: String
   def structure: String
+  def name: Name
+  def fullName: Ref
 }
 
 object Symbol {
+  private def unsupported(sym: Symbol, op: String) = {
+    val receiver = if (sym == None) "Symbol.None" else sym.syntax
+    throw new SemanticException(Position.None, s"Symbol.${op} not supported for $receiver")
+  }
+
   @none object None extends Symbol {
+    override def toString = syntax
     override def syntax = s""
     override def structure = s"""Symbol.None"""
-    override def toString = syntax
+    override def name: Name = unsupported(this, "name")
+    override def fullName: Ref = unsupported(this, "fullName")
   }
 
   @leaf class Local(addr: Address, start: Int, end: Int) extends Symbol {
+    override def toString = syntax
     override def syntax = s"${addr.syntax}@$start..$end"
     override def structure = s"""Symbol.Local(${addr.structure}, $start, $end)"""
-    override def toString = syntax
+    override def name: Name = ???
```

Comment:
Aha, because it's "not yet implemented" as opposed to "it won't be implemented". Nevermind